spec/helpers/mock_project.rb

```diff
@@ -0,0 +1,83 @@
+# typed: strict
+# frozen_string_literal: true
+
+require "open3"
+require "helpers/mock_gem"
+
+module Tapioca
+  # A mock project used for testing purposes
+  class MockProject < MockDir
+    extend T::Sig
+
+    # Path to Tapioca's source files
+    TAPIOCA_PATH = T.let((Pathname.new(__FILE__) / ".." / ".." / "..").to_s, String)
+
+    # Write `contents` to the gemfile in this project
+    sig { params(contents: String, append: T::Boolean).void }
+    def gemfile(contents, append: false)
+      write("Gemfile", contents, append: append)
+    end
+
+    # Add a gem requirement to this project's gemfile from a `MockGem`
+    sig { params(gem: MockGem).void }
+    def require_mock_gem(gem)
+      gemfile("#{gem.gemfile_line}\n", append: true)
+    end
+
+    # Add a gem requirement to this project's gemfile from a real gem
+    sig { params(name: String, version: T.nilable(String)).void }
+    def require_real_gem(name, version = nil)
+      line = String.new
+      line << "gem \"#{name}\""
+      line << ", \"#{version}\"" if version
+      line << "\n"
+      gemfile(line, append: true)
+    end
+
+    # Default Gemfile contents requiring only Tapioca
+    sig { returns(String) }
+    def tapioca_gemfile
+      <<~GEMFILE
+        source("https://rubygems.org")
+
+        gem "tapioca", path: "#{TAPIOCA_PATH}"
+      GEMFILE
+    end
+
+    # Write `contents` to the `sorbet/config` file of this project
+    sig { params(contents: String, append: T::Boolean).void }
+    def sorbet_config(contents, append: false)
+      write("sorbet/config", contents, append: append)
+    end
+
+    # Run `bundle install` in this project context (unbundled env)
+    sig { returns([String, String, T::Boolean]) }
+    def bundle_install
+      opts = {}
+      opts[:chdir] = path
+      Bundler.with_unbundled_env do
+        out, err, status = Open3.capture3("bundle", "install", opts)
+        [out, err, T.must(status.success?)]
+      end
+    end
+
+    # Run a `command` with `bundle exec` in this project context (unbundled env)
+    sig { params(command: String).returns([String, String, T::Boolean]) }
+    def bundle_exec(command)
+      opts = {}
+      opts[:chdir] = path
+      Bundler.with_unbundled_env do
+        out, err, status = Open3.capture3(["bundle", "exec", command].join(" "), opts)
+        [out, err, T.must(status.success?)]
+      end
+    end
+
+    # Run a Tapioca `command` with `bundle exec` in this project context (unbundled env)
+    sig { params(command: String).returns([String, String, T::Boolean]) }
+    def tapioca(command)
+      exec_command = ["tapioca", command]
+      exec_command << "--workers=1" if command.start_with?(/dsl|gem/) && !command.match?("--workers")
+      bundle_exec(exec_command.join(" "))
```

Comment:
Why do we need to run with one worker in tests?

Reply:
Because the output order changes when running with multiple forks and a lot of our tests expect a specific order in the output. It would require changing the expectations.

Reply:
> the output order changes when running with multiple forks

Maybe we should change this in subsequent PRs, I'm sure we can get the tests to run a bit faster with multiple workers.